syntax = "proto3";

option java_multiple_files = true;
option java_package = "nl.tno.simulation.externalmodel.protos";
option java_outer_classname = "Healthcheck";

enum HealthStatus {
    HEALTHY = 0;
    UNHEALTHY = 1;
}

message PingHealthSOToMSO {
    HealthStatus healthy = 1;
    repeated string reasons = 2; // Why is SO unhealthy?
    repeated string activeSimulations = 3;
}

message PongHealthMSOToSO {
    HealthStatus healthy = 1;
    repeated string reasons = 2; // Why is MSO unhealthy?
}

message PingHealthMSOToModel {
    HealthStatus healthy = 1;
    repeated string reasons = 2; // Why is MSO unhealthy?
}

message PongHealthModelToMSO {
    HealthStatus healthy = 1;
    repeated string reasons = 2; // Why is model unhealthy?
    int32 numberOfBids = 3;  // How many bids have been generated by model.
    int32 numberOfAllocations = 4;  // How many allocations have been allocated by model.
}
